docs/ingestion/schema-design.md

```diff
@@ -241,12 +241,11 @@ You should query for the number of ingested rows with:
 ]
 ```
 
-### Schema-less dimensions
+### Schema auto-discovery for dimensions
 
-If the `dimensions` field is left empty in your ingestion spec, Druid will treat every column that is not the timestamp column,
-a dimension that has been excluded, or a metric column as a dimension.
+You can have Druid infer the schema for your data partially or fully by setting `dimensionsSpec.useSchemaDiscovery` to `true` and defining some or no dimensions in the dimensions list. For any dimensions that aren't a uniform type, Druid ingests them as JSON.
 
-Note that when using schema-less ingestion, all dimensions will be ingested as String-typed dimensions.
+If you do not set `dimensionsSpec.useSchemaDiscovery` to `true` but do set `includeAllDimensions` to `true` with an empty dimensions list, Druid will ingest all columns as strings.
```

Comment:
Druid will also do the older string based schema discovery by just leaving the dimensions list empty. `includeAllDimensions` allows the string based mode to _also_ work with partial schema declaration, otherwise the only way to enable this mode is with an empty dimension list (and not setting the new `useSchemaDiscovery` to true)